design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+# Secrets Provider Phase 2 Milestone 1 - Solution Design
+
+## Table of Contents
+
+* [Glossary](#glossary)
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+    + [Motivation](#motivation)
+  * [Requirements](#requirements)
+    + [Milestone 1 *(current)*](#milestone-1---current--)
+    + [Milestone 2](#milestone-2)
+  * [Solution](#solution)
+    + [Milestone 1: Serve K8s secrets to multiple applications once (no rotation)](#milestone-1--serve-k8s-secrets-to-multiple-applications-once--no-rotation-)
+    + [Design](#design)
+      - [How does a Job configured via Helm answer the requirements?](#how-does-a-job-configured-via-helm-answer-the-requirements-)
+      - [What drawbacks does this solution have?](#what-drawbacks-does-this-solution-have-)
+      - [Customer experience](#customer-experience)
+      - [Packaging Helm Charts](#packaging-helm-charts)
+    + [Milestone 2: Rotation](#milestone-2--rotation)
+    + [Lifecycle](#lifecycle)
+      - [Milestone 1](#milestone-1)
+        * [Installation](#installation)
+        * [Lifecycle](#lifecycle-1)
+        * [Update](#update)
+        * [Delete](#delete)
+    + [Order of Deployment](#order-of-deployment)
+      - [Milestone 1](#milestone-1-1)
+    + [Backwards compatibility](#backwards-compatibility)
+      - [Milestone 1](#milestone-1-2)
+    + [Performance](#performance)
+      - [Milestone 1](#milestone-1-3)
+    + [Affected Components](#affected-components)
+  * [Security](#security)
+  * [Test Plan](#test-plan)
+    + [Integration tests](#integration-tests)
+    + [Performance tests](#performance-tests)
+  * [Logs](#logs)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Glossary
+
+| **Term**                                                     | **Description**                                              |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| [Job](*https://kubernetes.io/docs/concepts/workloads/controllers/job/*) | K8s entity that runs a pod for one-time operation. Once the pod exists, the Job terminates |
```

Comment:
Once the pods exists? Once a pods exits?

Reply:
the pod that the Job brings up

Reply:
so the Job terminates when the pod is up? Isn't the pod part of the Job?

Reply:
they terminate together

Reply:
so `exits` seems more fitting

Reply:
They terminate together but are not deleted.

See here: https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-termination-and-cleanup Specifically,
> When a Job completes, no more Pods are created, but the Pods are not deleted either